src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThread.js

```diff
@@ -55,16 +60,52 @@ const AnnotationThread = ({
         token,
     });
 
+    const [mentionSelectorContacts, setMentionSelectorContacts] = React.useState([]);
+
+    const getAvatarUrl = async (userId: string): Promise<?string> =>
+        api.getUsersAPI(false).getAvatarUrlWithAccessToken(userId, file.id);
+
+    const getMentionContactsSuccessCallback = (collaborators: { entries: SelectorItems<> }): void => {
+        setMentionSelectorContacts(collaborators.entries);
+    };
+
+    const getMentionWithQuery = debounce(searchStr => {
+        api.getFileCollaboratorsAPI(false).getCollaboratorsWithQuery(
+            file.id,
+            getMentionContactsSuccessCallback,
+            onError,
+            searchStr,
+        );
+    }, DEFAULT_COLLAB_DEBOUNCE);
+
     return (
         <div className={classNames('AnnotationThread', className)} data-testid="annotation-thread">
             <IntlProvider locale={language} messages={messages}>
-                <AnnotationThreadContent
-                    api={api}
-                    annotationId={annotationId}
-                    fileId={fileId}
-                    filePermissions={filePermissions}
-                    onError={onError}
-                />
+                {!annotationId ? (
```

Comment:
we could also make props variable based on if the annotationId exists and spread it on `AnnotationThreadCreate`

Reply:
I did not noticed this pattern to be commonly used in BUIE 🤔  Can you elaborate a little bit more, what would we gain with that approach? @JChan106